### ISSUE TITLE
Secrets not generated from controller output

### ISSUE THREAD

**Issue by timstahlappvia** [appvia/terraform-controller#102]
When deploying a configuration, the controller will generate the cloud resources (in this case a storage account in Azure), but will fail to create the secrets from the module output. The following error is displayed in the logs of the terraform-controller pod after deployment:

```
{"generation":"1","level":"debug","msg":"found the pod","name":"storageacct","namespace":"demo","pod":"storageacct-apply-kw566-p45cl","stage":"apply","time":"2022-06-03T12:14:24Z","uid":"7cbd4c87-58d8-45e5-ada9-5e45ab31a3e7"}
{"bytes":17718,"fields.time":48117799337,"ip":"10.170.0.102:52240","level":"info","method":"GET","msg":"received api request","response":200,"time":"2022-06-03T12:15:06Z"}
{"error":"json: cannot unmarshal array into Go struct field StateProperty.outputs.type of type string","level":"error","msg":"failed to reconcile the configuration resource","time":"2022-06-03T12:15:06Z"}
{"error":"json: cannot unmarshal array into Go struct field StateProperty.outputs.type of type string","level":"error","msg":"failed to reconcile the configuration resource","time":"2022-06-03T12:15:06Z"}
{"error":"json: cannot unmarshal array into Go struct field StateProperty.outputs.type of type string","level":"error","msg":"failed to reconcile the configuration resource","time":"2022-06-03T12:15:06Z"}
[...]
```
The following is the output from kubectl logs apply pod:
```
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

encryption_scope_ids = {}
id = "/subscriptions/b23a8772-031c-4494-a003-daf46b189839/resourceGroups/legendary-robot/providers/Microsoft.Storage/storageAccounts/appviastor3"
name = "appviastor3"
primary_access_key = <sensitive>
primary_blob_connection_string = <sensitive>
primary_blob_endpoint = "https://appviastor3.blob.core.windows.net/"
primary_blob_host = "appviastor3.blob.core.windows.net"
primary_connection_string = <sensitive>
primary_dfs_endpoint = "https://appviastor3.dfs.core.windows.net/"
primary_file_endpoint = "https://appviastor3.file.core.windows.net/"
primary_queue_endpoint = "https://appviastor3.queue.core.windows.net/"
primary_table_endpoint = "https://appviastor3.table.core.windows.net/"
primary_web_endpoint = "https://appviastor3.z13.web.core.windows.net/"
principal_id = "f071c56c-a68c-45cb-af28-2f757371ed03"
sa = <sensitive>
secondary_access_key = <sensitive>
secondary_blob_connection_string = <sensitive>
secondary_connection_string = <sensitive>
tenant_id = "7a770e35-b455-4df2-a276-b07408438d9a"
time="2022-06-03T12:15:05Z" level=info msg="successfully executed the step"
[build] completed
```

**Comment by timstahlappvia** [comment]
As a side -- it appears that even if you use the keys: fields in the configuration yaml to denote which keys you'd like (so I can skip that array at the beginning that it's most likely mad about) you get the exact same situation. Looks like it's trying to iterate over the entire list prior to writing the keys -- might be getting stopped at that top one that isn't a simply key:value pair.

**Comment by gambol99** [comment]
Closing as fixed ... will be available in [v0.1.6](https://github.com/appvia/terraform-controller/milestone/5)

**Comment by timstahlappvia** [comment]
Awesome, thanks!!
